app/src/main/java/org/oppia/app/player/state/audio/LanguageDialogFragment.kt

```diff
@@ -0,0 +1,69 @@
+package org.oppia.app.player.state.audio
+
+import android.app.Dialog
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import org.oppia.app.R
+import java.util.ArrayList
+
+/**
+ * DialogFragment that controls language selection in audio and written translations
+ */
+class LanguageDialogFragment : DialogFragment() {
+  companion object {
+    /**
+     * This function is responsible for displaying content in DialogFragment
+     * @param title: title of the Dialog
+     * @param languageCodeList: List of strings containing languages
+     * @param currentLanguageCode: Currently selected language code
+     * @return LanguageDialogFragment: DialogFragment
+     */
+    fun newInstance(
+      title: String,
+      languageCodeList: List<String>,
+      currentLanguageCode: String
+    ): LanguageDialogFragment {
+      val selectedIndex: Int = languageCodeList.indexOf(currentLanguageCode)
+      val frag = LanguageDialogFragment()
+      val args = Bundle()
+      args.putString("TITLE", title)
+      args.putStringArrayList(
+        "LANGUAGE__LIST",
+        convertLanguageCodeListToNameList(languageCodeList) as ArrayList<String>
+      )
+      args.putInt("CURRENT_LANGUAGE", selectedIndex)
+      frag.arguments = args
+      return frag
+    }
+
+    private fun convertLanguageCodeListToNameList(languageCodeList: List<String>): List<String> {
```

Comment:
Why not do this in the domain layer, instead? That seems a bit more fitting to me since it helps this fragment avoid needing to make any decisions about the input--all it should be responsible for is just displaying data and piping user responses back.

Reply:
Got your point. But for now I am removing this method so that we do not get blocked on work. And if the Language Alert Dialog is shown it will show `en` and `hi` rather than `English` and `Hindi`. Which will be solved in domain layer later on.